### PR TITLE
Spec aptos account

### PR DESCRIPTION
**Pull request by UIZorrot**
### Description
This is a part of spec work for Aptos Framework from MoveBit. The changes are:

Completed all specs left as TODO in aptos_account;
Some functions in code gets specified.

### Test Plan
`aptos move prove -f aptos_account.move`

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/aptos_account.move [hunk 1]
```diff
@@ -53,12 +53,35 @@ module aptos_framework::aptos_account {
...
+        }) {
             let to = *vector::borrow(&recipients, i);
             let amount = *vector::borrow(&amounts, i);
             transfer(source, to, amount);
+            spec {
+                assume amounts[i] == 0;
+            };
             i = i + 1;
-        };
+        };
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_account.move:84]: Remove the trailing spaces.

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move [hunk 2]
```diff
@@ -1,6 +1,6 @@
 spec aptos_framework::aptos_account {
     spec module {
-        pragma verify = true;
+        pragma verify=true;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:3]: This line can be removed.

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move [hunk 3]
```diff
@@ -47,8 +73,47 @@ spec aptos_framework::aptos_account {
     }
 
     spec batch_transfer(source: &signer, recipients: vector<address>, amounts: vector<u64>) {
-        // TODO: missing aborts_if spec
-        pragma verify=false;
+        pragma verify=true;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:76]: This line can be removed.

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move [hunk 4]
```diff
@@ -60,17 +125,121 @@ spec aptos_framework::aptos_account {
     }
 
     spec batch_transfer_coins<CoinType>(from: &signer, recipients: vector<address>, amounts: vector<u64>) {
-        // TODO: missing aborts_if spec
-        pragma verify=false;
+        use aptos_std::type_info;
+        pragma verify=true;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:129]: This line can be removed.

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move [hunk 5]
```diff
@@ -60,17 +125,121 @@ spec aptos_framework::aptos_account {
...
+        aborts_if !coin::is_account_registered<CoinType>(to) && !type_info::spec_is_struct<CoinType>();    
+        aborts_if !coin::is_account_registered<CoinType>(to) && !can_receive_direct_coin_transfers(to);
+
     }
 
     spec transfer_coins<CoinType>(from: &signer, to: address, amount: u64) {
-        // TODO: missing aborts_if spec
-        pragma verify=false;
+        use aptos_std::type_info;
+        pragma verify=true;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:210]: Remove this.

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move [hunk 6]
```diff
@@ -60,17 +125,121 @@ spec aptos_framework::aptos_account {
...
+        aborts_if exists i in 0..len(recipients):  
+            !coin::is_account_registered<CoinType>(recipients[i]) && !can_receive_direct_coin_transfers(recipients[i]);
+
     }
 
     spec deposit_coins<CoinType>(to: address, coins: Coin<CoinType>) {
-        // TODO: missing aborts_if spec
-        pragma verify=false;
+        use aptos_std::type_info;
+        pragma verify=true;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:184]: Remove this line.

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move [hunk 7]
```diff
@@ -28,6 +28,32 @@ spec aptos_framework::aptos_account {
 
     spec transfer(source: &signer, to: address, amount: u64) {
         pragma verify = false;
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:30]: why `verify=false`?

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move (hunk 6 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:184]: Should I remove use aptos_std::type_info? it seemed affected the result

File: aptos-move/framework/aptos-framework/sources/aptos_account.spec.move (hunk 7 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/aptos_account.spec.move:30]: it was a mistake, it will be true

**Review by junkil-park — COMMENTED** [review]
Can you define and use schemas? For example, the entire "guid properties" repeatedly appear in many places. You can define a schema and reuse it.

**Comment by UIZorrot** [comment]
> Can you define and use schemas? For example, the entire "guid properties" repeatedly appear in many places. You can define a schema and reuse it.

I have tried to fixed the problem mentioned above

**Review by junkil-park — APPROVED** [review]
LGTM!

**Review by rahxephon89 — APPROVED** [review]

**Review by movekevin — APPROVED** [review]

**Pull request merged**